packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -144,6 +145,40 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
                 }
         )
     }
+
+    @VisibleForTesting
+    fun onDeleteInstance(flutterResult: Result, request: HashMap<String, Any>) {
+        try {
+            var modelName = request["modelName"] as String
+            var modelData = request["model"] as HashMap<String, Any>
+            var queryOptions: QueryOptions = QueryOptionsBuilder.fromSerializedMap(request)
+            val plugin = Amplify.DataStore.getPlugin("awsDataStorePlugin") as AWSDataStorePlugin
```

Comment:
Can you share this as a class variable between this and query?

Reply:
I think so - but I'll discuss with @Amplifiyer.

Reply:
It's a good idea, I couldn't do it in my last PR as I struggled to mock a lateinit in the time being. Let me know if you want to give it a shot on figuring out the unit tests.